streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -16,121 +16,239 @@
  */
 package org.apache.kafka.streams.processor.internals;
 
-import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.utils.LogContext;
 import org.apache.kafka.streams.StreamsConfig;
 import org.apache.kafka.streams.StreamsMetrics;
+import org.apache.kafka.streams.processor.StateStore;
 import org.apache.kafka.streams.processor.TaskId;
 import org.apache.kafka.streams.processor.internals.metrics.StreamsMetricsImpl;
 import org.apache.kafka.streams.processor.internals.metrics.ThreadMetrics;
+import org.slf4j.Logger;
 
+import java.util.Collection;
 import java.util.Collections;
 import java.util.Map;
 import java.util.Set;
 
 /**
  * A StandbyTask
  */
-public class StandbyTask extends AbstractTask {
+public class StandbyTask extends AbstractTask implements Task {
+    private final TaskId id;
```

Comment:
Why do we move `TaskId` from `AbstractTask` to `StandbyTask` and also `StreamTask` -- seem we are duplicating code -- every task has an ID? Similar for other class members? Why not following best practice of the OO model?

Reply:
Yeah I can move this to AbstractTask.